modules/survey_accounts/php/addsurvey.class.inc

```diff
@@ -77,6 +77,19 @@ class AddSurvey extends \NDB_Form
               array('CandID' => $error);
         }
 
+        $projectID = $db->pselect(
+            "SELECT ProjectID FROM session
+            WHERE CandID=:v_CandID",
+            array(
+                'v_CandID' => $values['CandID']
+            )
+        );
+        if (!\NDB_Factory::singleton()->user()->hasProject(reset($projectID[0]))) {
```

Comment:
This is just the first session of the candidate, not the one submitted from the addsurvey page. Isn't there a visit submitted to add the survey to?

Reply:
Candidates aren't only related to one project, candidates can have sessions at different projects. That's why the field is called "RegistrationProjectID" and not "ProjectID".

Reply:
Ah ok. In this case, I will add the visit label to the query, so that it checks for the project related specifically to this session. Thank you for the clarification!